### DISCUSSION TITLE
Set Part Origin to include Z Axis

### DISCUSSION THREAD

**Discussion by garyzz** [synman/Octoprint-Bettergrblsupport#94] [Q&A] [answered]
Z-axis coordinate saved when setting part origin? 

If set part origin is currently taking X and Y only then adding Z would be nice.

**Comment by synman** [comment, marked as answer]
It's in there....  Check out the example workflow in the wiki:  https://github.com/synman/Octoprint-Bettergrblsupport/wiki/CNC-Workflow-Example -- hint: you don't have to have a z-probe.

Further context:  you select the axis / axes you want to use when setting and navigating to an origin using the **Axis Selection bar**.  In your case it sounds like what you want to do is establish your Z starting position by jogging and then select the "Z" axis in the Axis Selection Bar, then lastly "Set Work Origin" for it.


![](https://user-images.githubusercontent.com/1299716/147916286-cfd84110-21fd-4bed-852b-800aad2cc897.png)

**Comment by garyzz** [comment]
Perfect. Yes, I had XY set instead of ALL, so Z did not set for me.  I do like how you can select specific axis in the origin.  Thanks so much for clarifying this point. Great stuff.
